packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -20,14 +24,31 @@ export default util.createRule<Options, MessageIds>({
     messages: {
       noVarReqs: 'Require statement not part of import statement.',
     },
-    schema: [],
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowPackageDotJson: {
+            type: 'boolean',
+          },
+        },
+      },
+    ],
   },
-  defaultOptions: [],
+  defaultOptions: [{ allowPackageDotJson: false }],
   create(context) {
+    const allowPackageDotJson = context.options[0]?.allowPackageDotJson;
     return {
       'CallExpression, OptionalCallExpression'(
         node: TSESTree.CallExpression | TSESTree.OptionalCallExpression,
       ): void {
+        if (
+          allowPackageDotJson &&
+          /package.json$/.test(node.arguments[0]?.value)
```

Comment:
@bradzacher should argument have `PrimaryExpression`  as well? at least `LiteralExpression` ?

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/typescript-estree/src/ts-estree/ts-estree.ts#L564

Reply:
`argument` is typed as `Expression`:

https://github.com/typescript-eslint/typescript-eslint/blob/cc70e4fbadd0b15fd6af913a2e1e2ddd346fa558/packages/typescript-estree/src/ts-estree/ts-estree.ts#L344-L362

Which includes `TSUnaryExpression`:

https://github.com/typescript-eslint/typescript-eslint/blob/cc70e4fbadd0b15fd6af913a2e1e2ddd346fa558/packages/typescript-estree/src/ts-estree/ts-estree.ts#L501-L506

Which includes `LeftHandSideExpression`:

https://github.com/typescript-eslint/typescript-eslint/blob/cc70e4fbadd0b15fd6af913a2e1e2ddd346fa558/packages/typescript-estree/src/ts-estree/ts-estree.ts#L383-L395

Which includes both `LiteralExpression` and `PrimaryExpression`.

Reply:
`LiteralExpression` does have a `value` prop right ! so why this is showing type error. 

> `##[error]@typescript-eslint/eslint-plugin: src/rules/no-var-requires.ts(47,51): error TS2339: Property 'value' does not exist on type 'Expression'.`

Reply:
because typescript is strict in its handling of properties on union types.

Whilst in a pure JS sense, the code you wrote is "correct" (because `value` will either be `string` or `undefined` in reality), TS does not allow it because there is not a `value` prop on every single type.

In order to use the `value` prop, you must first narrow the type:
```ts
          allowPackageDotJson &&
          node.arguments[0]?.type === AST_NODE_TYPES.Literal &&
          typeof node.arguments[0]?.value === 'string' &&
          /package.json$/.test(node.arguments[0].value)
```